tests/framework/db/ActiveRecordTest.php

```diff
@@ -561,6 +561,134 @@ public function testJoinWith()
         $this->assertEquals(2, $orders[0]->items[0]->category->id);
     }
 
+    /**
+     * @depends testJoinWith
+     */
+    public function testJoinWithDuplicate()
```

Comment:
Could you please use `@dataProvider` if it's possible or split this long method to small methods if not?

Reply:
I'll split this method

Reply:
This method contains eight test cases